qiskit_nature/operators/second_quantization/fermionic_op.py

```diff
@@ -0,0 +1,368 @@
+            # if char2 is one of `-`, `+` we pick up a phase when commuting it to the position
+            # of char1
+            if char2 in ["-", "+"]:
+                # Construct the string through which we have to commute
+                permuting_through = label1[i + 1:]
+                # Count the number of times we pick up a minus sign when commuting
+                ncommutations = permuting_through.count("+") + permuting_through.count(
+                    "-"
+                )
+                new_coeff *= (-1) ** ncommutations
+
+            # Check what happens to the symbol
+            new_char = mapping[char1 + char2]
+            if new_char == 0:
+                return "I" * len(label1), 0
+            new_label += cast(str, new_char)
+
+        return new_label, new_coeff
+
+    def add(self, other: "FermionicOp") -> "FermionicOp":
+        """Returns a `FermionicOp` representing the sum of the given base fermionic
+        operators.
+        """
+        if not isinstance(other, FermionicOp):
+            raise TypeError(
+                "Unsupported operand type(s) for +: 'FermionicOp' and "
+                "'{}'".format(type(other).__name__)
+            )
+
+        # Check compatibility (i.e. operators act on same register length)
+        if self.register_length != other.register_length:
+            raise TypeError("Incompatible register lengths for '+'.")
+
+        label1, coeffs1 = zip(*self.to_list())
+        label2, coeffs2 = zip(*other.to_list())
+
+        return FermionicOp(list(zip(label1 + label2, coeffs1 + coeffs2)))
+
+    def to_list(self) -> List[Tuple[str, complex]]:
+        """Getter for the operator_list of `self`"""
+        return list(zip(self._labels, self._coeffs))
+
+    @property
+    def register_length(self) -> int:
+        """Getter for the length of the fermionic register that the FermionicOp `self` acts
+        on.
+        """
+        return self._register_length
+
+    def adjoint(self) -> "FermionicOp":
+        """Returns the complex conjugate transpose (dagger) of `self`."""
+
+        dagger_map = {"+": "-", "-": "+", "I": "I", "N": "N", "E": "E"}
+        label_list = []
+        coeff_list = []
+        for label, coeff in zip(self._labels, self._coeffs):
+            conjugated_coeff = coeff.conjugate()
+
+            daggered_label = ""
+
+            for i, char in enumerate(label):
+                daggered_label += dagger_map[char]
+                if char in ["+", "-"]:
+                    permute_through = label[i + 1:]
+                    conjugated_coeff *= (-1) ** (
+                        permute_through.count("+") + permute_through.count("-")
+                    )
+
+            label_list.append(daggered_label)
+            coeff_list.append(conjugated_coeff)
+
+        return FermionicOp(list(zip(label_list, coeff_list)))
+
+    def reduce(self, atol: Optional[float] = None, rtol: Optional[float] = None) -> "FermionicOp":
```

Comment:
Here we are missing a docstring.

Reply:
Super class has the docstring. (maybe not enough)

Reply:
Sorry for my mistake. The doc is not generated correctly. I'll fix it.